File: Simple-Bank-System/Simple-Bank-System.py
class Bank:
    def __init__(self, balance):
        self.balance = balance

    def checkBank(self, account):
        return (
            isinstance(account, int)
            and account >= 1
            and account <= len(self.balance)
        )

    def transfer(self, account1, account2, money):
        if (not self.checkBank(account1)
                or not self.checkBank(account2)
                or self.balance[account1 - 1] < money):
            return False
        else:
            self.balance[account1 - 1] -= money
            self.balance[account2 - 1] += money
            return True

    def deposit(self, account, money):
        if not self.checkBank(account):
            return False
        else:
            self.balance[account - 1] += money
            return True

    def withdraw(self, account, money):
        if (not self.checkBank(account)
                or self.balance[account - 1] < money):
            return False
        else:
            self.balance[account - 1] -= money
            return True

balance = [0]
myBank = Bank(balance)

print(myBank.deposit(1, 1000000000000)) 
print(myBank.transfer(1, 1, 1000000000000))  
print(myBank.withdraw(1, 1000000000000))  
print(myBank.deposit(1, 1))               
print(myBank.transfer(1, 1, 0))           
print(myBank.withdraw(1, 0))              

File: Simple-Bank-System/Simple-Bank-System.js


var Bank = function(balance) {
    this.balance = balance
};

Bank.prototype.checkBank = function(account){
    return account >= 1 && account <= this.balance.length && !isNaN(account-1)
}

Bank.prototype.transfer = function(account1, account2, money) {
    if(!this.checkBank(account1) || !this.checkBank(account2) || this.balance[account1-1]<money ){
        return false
    }else{
        this.balance[account1-1] -= money
        this.balance[account2-1] += money
        return true
    }
};

Bank.prototype.deposit = function(account, money) {
    if(!this.checkBank(account)){
        return false
    }else{
        this.balance[account-1] += money
        return true
    }
};

Bank.prototype.withdraw = function(account, money) {
    if(!this.checkBank(account) || this.balance[account-1]<money){
        return false
    }else{
        this.balance[account-1] -= money
        return true
    }
};

let balance = [0];
let myBank = new Bank(balance)

console.log(myBank.deposit(1,1000000000000))
console.log(myBank.transfer(1,1,1000000000000))
console.log(myBank.withdraw(1,1000000000000))
console.log(myBank.deposit(1,1))
console.log(myBank.transfer(1,1,0))
console.log(myBank.withdraw(1,0))